liferay-workspace/tests/blade.samples.integration.test/src/testIntegration/java/com/liferay/blade/samples/test/BladeSamplesTest.java

```diff
@@ -109,24 +109,47 @@ public void testAllBladeSamples() throws Exception {
 		}
 
 		for (String sampleBundleFile : bladeSampleOutputFiles) {
-			String installBundleOutput = BladeCLIUtil.installBundle(
-				new File(sampleBundleFile));
-
-			String printFileName = new File(sampleBundleFile).getName();
-
-			bundleIDAllMap.put(installBundleOutput, printFileName);
-
-			try (Jar jar = new Jar(sampleBundleFile, sampleBundleFile)) {
-				if (jar.getManifest().getMainAttributes().getValue(
-						"Fragment-Host") == null) {
-
-					bundleIDStartMap.put(installBundleOutput, printFileName);
+			String printFileName;
+			String installBundleOutput;
+			
+			if (sampleBundleFile.endsWith(".war")) {
+				printFileName = new File(sampleBundleFile).getName();
+
+				String output = BladeCLIUtil.execute(
+					"sh", "install", "webbundle:file://" + sampleBundleFile + 
+					"?Web-ContextPath=/" + printFileName);
+				
+				installBundleOutput = output.substring(
+					output.indexOf("bundle id:") + 11,
+					output.indexOf("\n",
+					output.indexOf("bundle id:")));		
+				
+				bundleIDAllMap.put(installBundleOutput, printFileName);
+				bundleIDStartMap.put(installBundleOutput, printFileName);
+			}
+			
+			else {
+				printFileName = new File(sampleBundleFile).getName();
```

Comment:
Set the `new File(sampleBundleFile)` to a new variable `sampleBundleFile`

Then make sure that `sampleBundleFile` exists.  If it exists, then on 139 do this `Jar jar = new Jar(sampleBundleFile)`